internal/kojiapi/openapi.yml

```diff
@@ -0,0 +1,174 @@
+openapi: 3.0.1
+info:
+  version: '1'
+  title: OSBuild Composer - Koji
+  description: Service to build and push images to Koji.
+  license:
+    name: Apache 2.0
+    url: 'https://www.apache.org/licenses/LICENSE-2.0.html'
+paths:
+  '/compose/{id}':
+    get:
+      summary: The status of a compose
+      parameters:
+        - in: path
+          name: id
+          schema:
+            type: string
+            format: uuid
+            example: 123e4567-e89b-12d3-a456-426655440000
+          required: true
+          description: ID of compose status to get
+      description: 'Get the status of a running or finished compose. This includes whether or not it succeeded, and also meta information about the result.'
+      responses:
+        '200':
+          description: Compose status
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ComposeStatus'
+        '400':
+          description: Invalid compose id
+          content:
+            text/plain:
+              schema:
+                type: string
+        '404':
+          description: Unknown compose id
+          content:
+            text/plain:
+              schema:
+                type: string
+  /compose:
+    post:
+      summary: Create compose
+      description: 'Create a new compose, potentially consisting of several images and upload each to koji.'
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ComposeRequest'
+      responses:
+        '201':
+          description: Compose has started
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ComposeResponse'
+        '400':
+          description: Invalid compose request
+          content:
+            text/plain:
+              schema:
+                type: string
+        '415':
+          description: The content type is not supported
+          content:
+            text/plain:
+              schema:
+                type: string
+        '500':
```

Comment:
Is it necessary to list this return code explicitly? In such case, why not to include it in the other endpoint as well?

Reply:
I only included the return codes we emit. I think in this case we may want to use a different one than 500, but wasn't entirely sure so left it for now. This is triggered if depsolving fails. Not really an internal server error, just could be a million different things.

Reply:
On further reflection, I think we should blame the caller for this and return a bad request. I'll update.

Reply:
Ugh, I'm talking nonsense. We return this if enqueue fails. Maybe @larskarlitski can shed some light on hat is the right thing to return in that case.

Reply:
I went with a panic, as this would be a programming error.

Reply:
I'm not sure how chi works, but panicking in an endpoint handler function results in HTTP 500, no? Therefore I think it is "implicit" that 500 can happen. Anyway I was mostly curious, it's not super important.

Reply:
It's using `http.Server` as well, so it will return 500 on panics indeed.

I think panicking when `Enqueue` fails should be ok.